src/others/components/map/Map.tsx

```diff
@@ -31,12 +35,23 @@ const initialUkraineCenterView = {
 
 const MAP_STYLE = process.env.REACT_APP_MAPLIBRE_MAP_STYLE || "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json";
 
-export const Map = ({ sourceWithLayer, interactiveLayerIds }: MapProps) => {
+export const Map = ({ requestMapDataPoints }: MapProps) => {
   const { t } = useTranslation();
   const mapRef = useRef<MapRef>(null);
+  const { translateLocation } = useDictionaryContext();
   const [popupInfo, setPopupInfo] = useState<PopupInfo | null>(null);
   const [cursor, setCursor] = useState<"auto" | "pointer">("auto");
 
+  const searchParams = new URLSearchParams(window.location.search);
+  const showRegions = (searchParams.get("show_regions") || process.env.REACT_APP_SHOW_REGIONS) === "true";
+  
+  const interactiveLayerIds = useMemo(() => {return showRegions ? ["circles", "state-fills"] : ["circles"];}, [showRegions]);
```

Comment:
**issue**
`useMemo` is designed to use with expensive computations: https://reactjs.org/docs/hooks-reference.html#usememo, which this is not, so I think it can be removed.

Reply:
In short, it's an overkill and it will work just fine if you do: 

```ts
const interactiveLayerIds = showRegions ? ["circles", "state-fills"] : ["circles"]
```

Reply:
Yeah, exactly. To avoid rerendering the array reference needs to be constant. That's why I added the useMemo.